internal/transport/http2_client.go

```diff
@@ -1007,6 +1019,9 @@ func (t *http2Client) GracefulClose() {
 	active := len(t.activeStreams)
 	t.mu.Unlock()
 	if active == 0 {
+		if logger.V(logLevel) {
+			logger.Infof("transport: closing transport because no active streams left to process in GracefulClose call.")
+		}
```

Comment:
Fair enough about sticking this information in err you pass into t.Close(). Logging every time GracefulClose() is called SGTM, but I disagree with "This is a sub-case but maybe not a necessary one since it's pretty easy to determine if we know GracefulClose was called", in master, but I think it will be easy if you populate err and log in t.Close(), which I'll go ahead and do.